### PR TITLE
update cert-manager guides for Coder v1.21 (TLS for root domain + dev URLs)

### PR DESCRIPTION
**Pull request by bpmct**
Background: New installs on Coder v1.21 no longer contain an ingress by default. This means `Certificate` objects have to be manually created and referenced in the helm charts.

## Tested with
✅ CloudFlare✅
✅ Cloud DNS
❌ route53
❌ Azure DNS

If anybody has a deployment using these providers, let me know. If not, I'll probably test at least one to ensure that the certificate objects work fine with a `ClusterIssuer`. Other than that, the diffs in this PR are minor across providers

### CODE REVIEW

File: admin/devurls.md [hunk 1]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
 
 ```yaml
  - host: "*.my-custom-domain.io"
     http:
       paths:
       - path: /
         backend:
-          serviceName: envproxy
+          serviceName: coderd
           servicePort: 8080
```
Comment (bpmct) [inline admin/devurls.md:41]: @kylecarbs I'm unsure if this port is right and I don't believe we have any docs on providing an ingress. Should I omit this for now?

File: admin/devurls.md [hunk 2]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
 ```shell
-helm upgrade coder coder/coder --set devurls.host="*.my-custom-domain.io"
+helm upgrade coder coder/coder --set coderd.devurlsHost="*.my-custom-domain.io"
 ```
 
-If you're using the default ingress controller, specifying a value for
-`devurls.host` automatically adds a rule that routes dev URL requests to the
-user's workspace:
+**Note:** If you are providing an ingress controller, then you will need to add
+the rule manually to the ingress. If you are unsure, skip this step:
```
Comment (khorne3) [inline admin/devurls.md:32]:
```suggestion
the rule manually to the ingress.
```

File: admin/devurls.md [hunk 3]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
-
 ### Step 2: Modify the wildcard DNS record
 
 The final step to enabling dev URLs is to update your wildcard DNS record. Get
-the **ingress IP address** using `kubectl --namespace coder get ingress` and
-point your wildcard DNS record (e.g., \*.my-custom-domain.io) to the ingress IP
-address.
+the **LoadBalancer IP address** using `kubectl --namespace coder get svc` and
+point your wildcard DNS record (e.g., \*.my-custom-domain.io) to the
+LoadBalancer's IP address.
```
Comment (khorne3) [inline admin/devurls.md:49]: Do we want to mention the line is `ingress-nginx`? I'm assuming it's the external IP address not the cluster-IP?

File: admin/devurls.md [hunk 4]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
+
+## Step 3 (Optional): Add a TLS certificate
+
+For secure (HTTPS) dev URLs, you can add (or generate) a TLS certificate for the
+wildcard domain.
+
+- Guides: [TLS with LetsEncrypt](../guides/ssl-certifcates)
```
Comment (khorne3) [inline admin/devurls.md:56]:
```suggestion
- See our [guide for creating a TLS certificate using LetsEncrypt](../guides/ssl-certifcates)
```

File: admin/devurls.md [hunk 5]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
+- To add a custom certificate, reference our
```
Comment (khorne3) [inline admin/devurls.md:57]:
```suggestion
- To add a custom certificate, refer to our
```

File: admin/devurls.md [hunk 6]
```diff
@@ -22,35 +22,40 @@ your cluster, you'll need to modify your:
...
+  [helm chart](https://github.com/cdr/enterprise-helm)
```
Comment (khorne3) [inline admin/devurls.md:58]:
```suggestion
  [Helm chart](https://github.com/cdr/enterprise-helm)
```

File: admin/devurls.md [hunk 7]
```diff
@@ -87,9 +92,9 @@ scroll down to **Dev URL Access Permissions**.
 You can set the maximum access level, but developers may choose to restrict
 access further.
 
-For example, if you set the maximum access level as
-**Authenticated**, then any dev URLs created for workspaces in your Coder
-deployment will be accessible to any authenticated Coder user.
+For example, if you set the maximum access level as **Authenticated**, then any
```
Comment (khorne3) [inline admin/devurls.md:95]:
```suggestion
For example, if you set the maximum access level as **Authenticated**, then all
```

File: admin/workspace-providers/deployment.md [hunk 8]
```diff
@@ -76,11 +76,11 @@ Install the following dependencies if you haven't already:
...
 
-   ```console
-   serviceaccount/coder created
-   role.rbac.authorization.k8s.io/coder created
-   rolebinding.rbac.authorization.k8s.io/coder created
-   ```
+   For installations using HTTPS, you must
+   [ensure the deployment has a valid certificate](../../guides/tls-certificates/index.md).
+
+   If you're unfamiliar with the helm configuration values file, see our doc on
```
Comment (khorne3) [inline admin/workspace-providers/deployment.md:82]:
```suggestion
   If you're unfamiliar with the Helm configuration values file, see our doc on
```

File: admin/workspace-providers/deployment.md [hunk 9]
```diff
@@ -76,11 +76,11 @@ Install the following dependencies if you haven't already:
...
+   [updating a helm chart](../../guides/admin/helm-charts.md)
```
Comment (khorne3) [inline admin/workspace-providers/deployment.md:83]:
```suggestion
   [updating a Helm chart](../../guides/admin/helm-charts.md)
```

File: guides/tls-certificates/cloudDNS.md [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
 ---
 
 [cert-manager](https://cert-manager.io/) allows you to enable HTTPS on your
 Coder installation, regardless of whether you're using
 [Let's Encrypt](https://letsencrypt.org/) or you have your own certificate
 authority.
 
-This guide will show you how to install cert-manager v1.0.1 and set up your
+> Note: This guide is for Coder v1.21 which has a slightly different way of
+> handling certificates. Ensure you are reading the docs for your Coder version.
```
Comment (khorne3) [inline guides/tls-certificates/cloudDNS.md:14]: `This guide is for Coder v1.21.0 and later, which handle certificates differently from earlier versions of Coder. Ensure that you're reading the docs applicable to your Coder version.`

File: guides/tls-certificates/cloudflare.md [hunk 11]
```diff
@@ -10,7 +10,10 @@ Coder installation, regardless of whether you're using
 [Let's Encrypt](https://letsencrypt.org/) or you have your own certificate
 authority.
 
-This guide will show you how to install cert-manager v1.0.1 and set up your
+> Note: This guide is for Coder v1.21 which has a slightly different way of
+> handling certificates. Ensure you are reading the docs for your Coder version.
+
```
Comment (khorne3) [inline guides/tls-certificates/cloudflare.md:15]: see suggested wording in cloudDNS.md

File: guides/tls-certificates/route53.md [hunk 12]
```diff
@@ -10,22 +10,22 @@ Coder installation, regardless of whether you're using
 [Let's Encrypt](https://letsencrypt.org/) or you have your own certificate
 authority.
 
-This guide will show you how to install cert-manager v1.0.1 and set up your
+> Note: This guide is for Coder v1.21 which has a slightly different way of
+> handling certificates. Ensure you are reading the docs for your Coder version.
+
```
Comment (khorne3) [inline guides/tls-certificates/route53.md:15]: see suggested wording in cloudDNS.md

**Comment by khorne3** [comment]
@bpmct fixed the release branch so it should be good to go w.r.t. `main`

**Comment by bpmct** [comment]
@khorne3 tested this on cloud DNS. i would say it is good to go

**Review by khorne3 — APPROVED** [review]

**Pull request merged**